src/tools/genpolicy/src/registry.rs

```diff
@@ -508,3 +628,31 @@ fn build_auth(reference: &Reference) -> RegistryAuth {
 
     RegistryAuth::Anonymous
 }
+
+fn parse_passwd_file(passwd: String) -> Result<Vec<PasswdRecord>> {
+    let mut records = Vec::new();
+
+    for rec in passwd.lines() {
+        let fields: Vec<&str> = rec.split(':').collect();
+
+        let field_count = fields.len();
+        if field_count != 7 {
+            return Err(anyhow!(
+                "Incorrect passwd record, expected 7 fields, got {}",
+                field_count
+            ));
+        }
+
+        records.push(PasswdRecord {
+            user: fields[0].to_string(),
+            validate: fields[1] == "x",
+            uid: fields[2].parse().unwrap(),
+            gid: fields[3].parse().unwrap(),
```

Comment:
Did you mean to use the question-mark operator?

Reply:
genpolicy panics on various paths when it deals with unexpected input format. So, the unwrap is consistent with the current design.